{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'url': ''}\n"
     ]
    }
   ],
   "source": [
    "import requests\n",
    "\n",
    "r = requests.post(\n",
    "\t\"http://localhost:8080/predict\", \n",
    "\tjson={\"url\":\"\"}\n",
    ")\n",
    "\n",
    "print(r.json())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yes\n"
     ]
    }
   ],
   "source": [
    "\n",
    "r = requests.post(\n",
    "\t\"http://localhost:8080/download_registry_model\", \n",
    "\tjson={\"model_name\":\"log_reg_shot_dist_and_angle\",\"version\":\"pp\"}\n",
    ")\n",
    "\n",
    "print(r.json())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO:app:{'qq': 'pp', 'qwdqwd': 123}\n",
      "ERROR:app:Exception on /predict [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 112, in predict\n",
      "    raise NotImplementedError(\"TODO: implement this enpdoint\")\n",
      "NotImplementedError: TODO: implement this enpdoint\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:01:43] \"\u001b[35m\u001b[1mPOST /predict HTTP/1.1\u001b[0m\" 500 -\n",
      "INFO:app:{'qq': 'pp'}\n",
      "ERROR:app:Exception on /predict [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 112, in predict\n",
      "    raise NotImplementedError(\"TODO: implement this enpdoint\")\n",
      "NotImplementedError: TODO: implement this enpdoint\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:02:27] \"\u001b[35m\u001b[1mPOST /predict HTTP/1.1\u001b[0m\" 500 -\n",
      "INFO:app:qq\n",
      "ERROR:app:Exception on /predict [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 112, in predict\n",
      "    raise NotImplementedError(\"TODO: implement this enpdoint\")\n",
      "NotImplementedError: TODO: implement this enpdoint\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:02:39] \"\u001b[35m\u001b[1mPOST /predict HTTP/1.1\u001b[0m\" 500 -\n",
      "INFO:app:qq\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:03:02] \"POST /predict HTTP/1.1\" 200 -\n",
      "INFO:app:{'qq': 'pp', 'qwdqwd': 123}\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:03:10] \"POST /predict HTTP/1.1\" 200 -\n",
      "INFO:app:{'url': ''}\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:04:03] \"POST /predict HTTP/1.1\" 200 -\n",
      "INFO:app:{'url': ''}\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:04:09] \"POST /predict HTTP/1.1\" 200 -\n",
      "INFO:app:{'url': ''}\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:04:15] \"POST /predict HTTP/1.1\" 200 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 53, in logs\n",
      "    raise NotImplementedError(\"TODO: implement this endpoint\")\n",
      "NotImplementedError: TODO: implement this endpoint\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:07:12] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:07:35] \"GET /logs HTTP/1.1\" 200 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:15:49] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:16:31] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:16:39] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:17:27] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:19:26] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:19:59] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:20:55] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:21:05] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:22:28] \"GET /logs HTTP/1.1\" 200 -\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:22:34] \"GET /logs HTTP/1.1\" 200 -\n",
      "INFO:app:{'url': ''}\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:22:40] \"POST /predict HTTP/1.1\" 200 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:22:56] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:23:39] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "ERROR:app:Exception on /logs [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 63, in logs\n",
      "    return jsonify(response)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 302, in jsonify\n",
      "    f\"{dumps(data, indent=indent, separators=separators)}\\n\",\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 132, in dumps\n",
      "    return _json.dumps(obj, **kwargs)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\__init__.py\", line 238, in dumps\n",
      "    **kw).encode(obj)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 199, in encode\n",
      "    chunks = self.iterencode(o, _one_shot=True)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 257, in iterencode\n",
      "    return _iterencode(o, 0)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\json\\__init__.py\", line 51, in default\n",
      "    return super().default(o)\n",
      "  File \"C:\\ProgramData\\miniconda3\\lib\\json\\encoder.py\", line 179, in default\n",
      "    raise TypeError(f'Object of type {o.__class__.__name__} '\n",
      "TypeError: Object of type bytes is not JSON serializable\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:23:57] \"\u001b[35m\u001b[1mGET /logs HTTP/1.1\u001b[0m\" 500 -\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:24:02] \"GET /logs HTTP/1.1\" 200 -\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:25:37] \"GET /logs HTTP/1.1\" 200 -\n",
      "INFO:app:logs read.\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:25:48] \"GET /logs HTTP/1.1\" 200 -\n",
      "INFO:app:{'url': ''}\n",
      "ERROR:app:Exception on /download_registry_model [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 97, in download_registry_model\n",
      "    model_name = json[\"model_name\"]\n",
      "KeyError: 'model_name'\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:53:16] \"\u001b[35m\u001b[1mPOST /download_registry_model HTTP/1.1\u001b[0m\" 500 -\n",
      "INFO:app:{'url': ''}\n",
      "ERROR:app:Exception on /download_registry_model [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 97, in download_registry_model\n",
      "    model_name = json[\"model_name\"]\n",
      "KeyError: 'model_name'\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:53:53] \"\u001b[35m\u001b[1mPOST /download_registry_model HTTP/1.1\u001b[0m\" 500 -\n",
      "INFO:app:{'url': 'ffff'}\n",
      "ERROR:app:Exception on /download_registry_model [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 2073, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1519, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1517, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\yujd1\\AppData\\Roaming\\Python\\Python310\\site-packages\\flask\\app.py\", line 1503, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)\n",
      "  File \"C:\\Users\\yujd1\\Documents\\GitHub\\datascience\\milestone3\\serving\\app.py\", line 97, in download_registry_model\n",
      "    model_name = json[\"model_name\"]\n",
      "KeyError: 'model_name'\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:54:57] \"\u001b[35m\u001b[1mPOST /download_registry_model HTTP/1.1\u001b[0m\" 500 -\n",
      "INFO:app:logs read.\n",
      "INFO:werkzeug:127.0.0.1 - - [03/Dec/2023 23:55:24] \"GET /logs?model=qq&version=pp HTTP/1.1\" 200 -\n",
      "\n"
     ]
    }
   ],
   "source": [
    "r = requests.get(\n",
    "\t\"http://localhost:8080/logs\", \n",
    "    {\"model_name\":\"qq\",\"version\":\"pp\"}\n",
    ")\n",
    "\n",
    "print(r.json())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "from comet_ml import API"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "api = API(\"v5q8O8LftZtvOcoXlVM8Ku8fH\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "#get the Model object\n",
    "model = api.get_model(workspace=\"hfereidouni\", model_name=\"log_reg_shot_dist_and_angle\")\n",
    "\n",
    "model.download(\"1.9.0\",output_folder=\"./models/\", expand=True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "json = {\"model_name\":1,\"version\":2,}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "api = API(\"v5q8O8LftZtvOcoXlVM8Ku8fH\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "print(os.environ.get('COMET_API_KEY'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>#sk-container-id-2 {color: black;}#sk-container-id-2 pre{padding: 0;}#sk-container-id-2 div.sk-toggleable {background-color: white;}#sk-container-id-2 label.sk-toggleable__label {cursor: pointer;display: block;width: 100%;margin-bottom: 0;padding: 0.3em;box-sizing: border-box;text-align: center;}#sk-container-id-2 label.sk-toggleable__label-arrow:before {content: \"▸\";float: left;margin-right: 0.25em;color: #696969;}#sk-container-id-2 label.sk-toggleable__label-arrow:hover:before {color: black;}#sk-container-id-2 div.sk-estimator:hover label.sk-toggleable__label-arrow:before {color: black;}#sk-container-id-2 div.sk-toggleable__content {max-height: 0;max-width: 0;overflow: hidden;text-align: left;background-color: #f0f8ff;}#sk-container-id-2 div.sk-toggleable__content pre {margin: 0.2em;color: black;border-radius: 0.25em;background-color: #f0f8ff;}#sk-container-id-2 input.sk-toggleable__control:checked~div.sk-toggleable__content {max-height: 200px;max-width: 100%;overflow: auto;}#sk-container-id-2 input.sk-toggleable__control:checked~label.sk-toggleable__label-arrow:before {content: \"▾\";}#sk-container-id-2 div.sk-estimator input.sk-toggleable__control:checked~label.sk-toggleable__label {background-color: #d4ebff;}#sk-container-id-2 div.sk-label input.sk-toggleable__control:checked~label.sk-toggleable__label {background-color: #d4ebff;}#sk-container-id-2 input.sk-hidden--visually {border: 0;clip: rect(1px 1px 1px 1px);clip: rect(1px, 1px, 1px, 1px);height: 1px;margin: -1px;overflow: hidden;padding: 0;position: absolute;width: 1px;}#sk-container-id-2 div.sk-estimator {font-family: monospace;background-color: #f0f8ff;border: 1px dotted black;border-radius: 0.25em;box-sizing: border-box;margin-bottom: 0.5em;}#sk-container-id-2 div.sk-estimator:hover {background-color: #d4ebff;}#sk-container-id-2 div.sk-parallel-item::after {content: \"\";width: 100%;border-bottom: 1px solid gray;flex-grow: 1;}#sk-container-id-2 div.sk-label:hover label.sk-toggleable__label {background-color: #d4ebff;}#sk-container-id-2 div.sk-serial::before {content: \"\";position: absolute;border-left: 1px solid gray;box-sizing: border-box;top: 0;bottom: 0;left: 50%;z-index: 0;}#sk-container-id-2 div.sk-serial {display: flex;flex-direction: column;align-items: center;background-color: white;padding-right: 0.2em;padding-left: 0.2em;position: relative;}#sk-container-id-2 div.sk-item {position: relative;z-index: 1;}#sk-container-id-2 div.sk-parallel {display: flex;align-items: stretch;justify-content: center;background-color: white;position: relative;}#sk-container-id-2 div.sk-item::before, #sk-container-id-2 div.sk-parallel-item::before {content: \"\";position: absolute;border-left: 1px solid gray;box-sizing: border-box;top: 0;bottom: 0;left: 50%;z-index: -1;}#sk-container-id-2 div.sk-parallel-item {display: flex;flex-direction: column;z-index: 1;position: relative;background-color: white;}#sk-container-id-2 div.sk-parallel-item:first-child::after {align-self: flex-end;width: 50%;}#sk-container-id-2 div.sk-parallel-item:last-child::after {align-self: flex-start;width: 50%;}#sk-container-id-2 div.sk-parallel-item:only-child::after {width: 0;}#sk-container-id-2 div.sk-dashed-wrapped {border: 1px dashed gray;margin: 0 0.4em 0.5em 0.4em;box-sizing: border-box;padding-bottom: 0.4em;background-color: white;}#sk-container-id-2 div.sk-label label {font-family: monospace;font-weight: bold;display: inline-block;line-height: 1.2em;}#sk-container-id-2 div.sk-label-container {text-align: center;}#sk-container-id-2 div.sk-container {/* jupyter's `normalize.less` sets `[hidden] { display: none; }` but bootstrap.min.css set `[hidden] { display: none !important; }` so we also need the `!important` here to be able to override the default hidden behavior on the sphinx rendered scikit-learn.org. See: https://github.com/scikit-learn/scikit-learn/issues/21755 */display: inline-block !important;position: relative;}#sk-container-id-2 div.sk-text-repr-fallback {display: none;}</style><div id=\"sk-container-id-2\" class=\"sk-top-container\"><div class=\"sk-text-repr-fallback\"><pre>LogisticRegression()</pre><b>In a Jupyter environment, please rerun this cell to show the HTML representation or trust the notebook. <br />On GitHub, the HTML representation is unable to render, please try loading this page with nbviewer.org.</b></div><div class=\"sk-container\" hidden><div class=\"sk-item\"><div class=\"sk-estimator sk-toggleable\"><input class=\"sk-toggleable__control sk-hidden--visually\" id=\"sk-estimator-id-2\" type=\"checkbox\" checked><label for=\"sk-estimator-id-2\" class=\"sk-toggleable__label sk-toggleable__label-arrow\">LogisticRegression</label><div class=\"sk-toggleable__content\"><pre>LogisticRegression()</pre></div></div></div></div></div>"
      ],
      "text/plain": [
       "LogisticRegression()"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import joblib\n",
    "from flask import Flask, jsonify, request, abort\n",
    "model = joblib.load('./models/Log_Reg_shot_dist_only.joblib')\n",
    "model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'LogisticRegression'"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(model).__name__"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "dictionary update sequence element #0 has length 9; 2 is required",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32mc:\\Users\\yujd1\\Documents\\GitHub\\datascience\\Milestone3\\serving\\test.ipynb 单元格 12\u001b[0m line \u001b[0;36m1\n\u001b[1;32m----> <a href='vscode-notebook-cell:/c%3A/Users/yujd1/Documents/GitHub/datascience/Milestone3/serving/test.ipynb#X14sZmlsZQ%3D%3D?line=0'>1</a>\u001b[0m jsonify(\u001b[39mdict\u001b[39;49m(\u001b[39mlist\u001b[39;49m(model\u001b[39m.\u001b[39;49mfeature_names_in_)))\n",
      "\u001b[1;31mValueError\u001b[0m: dictionary update sequence element #0 has length 9; 2 is required"
     ]
    }
   ],
   "source": [
    "jsonify(dict(list(model.feature_names_in_)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Response [500]>\n"
     ]
    }
   ],
   "source": [
    "r = requests.get(\n",
    "\t\"http://localhost:8080/check\",\n",
    ")\n",
    "\n",
    "print(r)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
